UIforETW/UIforETWDlg.cpp

```diff
@@ -51,13 +863,16 @@ void outputPrintf(_Printf_format_string_ const wchar_t* pFormat, ...)
 	va_end(args);
 }
 
-void CUIforETWDlg::vprintf(const wchar_t* pFormat, va_list args)
+void CUIforETWDlg::vprintf(PCWSTR pFormat, va_list args)
 {
 	wchar_t buffer[5000];
 	_vsnwprintf_s(buffer, _TRUNCATE, pFormat, args);
 
-	for (const wchar_t* pBuf = buffer; *pBuf; ++pBuf)
+	for (PCWSTR pBuf = buffer; *pBuf; ++pBuf)
 	{
+		ATLASSERT(pBuf < (buffer + ARRAYSIZE(buffer)));

```

Comment:
You know, I think "programming on quicksand" actually describes everything that's wrong with my current style of programming. It's like, hilariously accurate.